apps/vanilla/src/scripts/renderFilterByType.ts

```diff
@@ -0,0 +1,29 @@
+
+import { FILTER_TYPE_OPTIONS, KEY_ACTIVE_PAGE, KEY_TYPE } from '../constants';
+import { SearchParamsService } from '../services/searchParams';
+import { setDefaultSelected } from '../utils';
+
+import { getInitialQueryParams } from './initAnimeTable';
+
+import { renderListOnActivePage } from './renderPagination';
+
+const selectType = document.querySelector<HTMLSelectElement>('.filter__select_type');
+
+/** Render filter by type. */
+export function renderFilterByType(): void {
+  const typeOptionHTML = FILTER_TYPE_OPTIONS.map(item => `<option value="${item.title}">${item.title}</option>`).join('');
+  const searchParams = SearchParamsService.getSearchParams();
+  if (selectType == null) {
+    return;
+  }
+  selectType.innerHTML = typeOptionHTML;
+
+  setDefaultSelected(selectType, searchParams.type != null ? FILTER_TYPE_OPTIONS.filter(item =>
+    item.value === searchParams.type)[0].title : FILTER_TYPE_OPTIONS[0].title) ;
```

Comment:
I would format it this way.
```suggestion
  setDefaultSelected(
    selectType, searchParams.type != null ?
      FILTER_TYPE_OPTIONS.filter(item => item.value === searchParams.type)[0].title :
      FILTER_TYPE_OPTIONS[0].title
  );
```